src/components/Choice/src/ChoiceOption.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+	<button
+		:class="[
+			$s.Button,
+			{ [$s.selected]: isSelected },
+		]"
+		@click="selectSelf"
+	>
+		<slot />
+	</button>
+</template>
+
+<script>
+import key from './key';
+
+export default {
+	inject: {
+		controlState: key,
+	},
+
+	props: {
+		value: {
+			type: undefined,
+			required: true,
+		},
+	},
+
+	computed: {
+		isSelected() {
+			if (this.controlState.isMultiSelect) {
+				return this.controlState.currentValue.includes(this.value);
+			}
+			return this.controlState.currentValue === this.value;
+		},
+	},
+
+	methods: {
+		selectSelf() {
+			let currentValue = this.value;
+
+			if (this.controlState.isMultiSelect) {
+				const currentValueArray = this.controlState.currentValue.slice();
+
+				if (currentValueArray.includes(currentValue)) {
+					currentValueArray.splice(currentValueArray.indexOf(currentValue), 1);
+				} else {
+					currentValueArray.push(currentValue);
+				}
+
+				currentValue = currentValueArray;
+			}
+
+			this.controlState.currentValue = currentValue;
```

Comment:
I'm not entirely sure that any props watching `controlState` will respond to this value being set if it's deeply nested. Probably not an issue unless you intend for `controlState` to be watched in this way.